himbaechel/uarch/gowin/gowin.cc

```diff
@@ -81,16 +81,19 @@ void GowinImpl::init_database(Arch *arch)
     if (args.options.count("family")) {
         family = args.options.at("family");
     } else {
-        bool GW2 = args.device == "GW2A-LV18PG256C8/I7";
+        bool GW2 = args.device.rfind("GW2A", 0) == 0;
```

Comment:
nitpick - why `rfind` and not `find`?

Reply:
find() with pos parameter only looks in that position or later.
rfind()  with pos parameter only looks in that position or earlier.
if pos == 0 then rfind is faster?

Reply:
Right, makes sense, thanks for explaining